modules/openapi-generator/src/main/resources/r/api.mustache

```diff
@@ -543,24 +547,21 @@
       if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
         {{#returnType}}
         {{#isPrimitiveType}}
-        local_var_content <- local_var_resp$response
-          local_var_resp, "text", encoding = "UTF-8", simplifyVector = FALSE
-        )
         # save response in a file
         if (!is.null(data_file)) {
-          write(local_var_content, data_file)
+          private$WriteFile(local_var_resp, data_file)
         }
 
         ApiResponse$new(content,resp)
         {{/isPrimitiveType}}
         {{^isPrimitiveType}}
         # save response in a file
         if (!is.null(data_file)) {
-          write(local_var_resp$response, data_file)
+          private$WriteFile(local_var_resp, data_file)
         }
 
         deserialized_resp_obj <- tryCatch(
-          self$api_client$deserialize(local_var_resp$response_as_text(), "{{returnType}}", loadNamespace("{{packageName}}")),
+          private$Deserialize(local_var_resp, "{{returnType}}"),
```

Comment:
`private$Deserialize()` fixes a bug. There was an assumption that all responses could be deserialized. This is not the case (again, the gzipped directory example). The private method will attempt to replicate the previous behavior, but will fall back on returning the raw response vector.